tests/rptest/tests/prealloc_nodes.py

```diff
@@ -17,32 +17,57 @@ class PreallocNodesTest(RedpandaTest):
     """
     Extends RedpandaTest to preallocate ducktape
     nodes from the given test context.
+
+    Having some explicitly allocated nodes is useful for
+    re-using those nodes to run multiple services at once, or to
+    run services that need to be on the same host (e.g. the FranzGoVerifiable
+    producer/consumers that coordinate via a local file)
     """
     def __init__(self, test_context: TestContext, node_prealloc_count: int,
                  *args, **kwargs):
         super(PreallocNodesTest, self).__init__(test_context, *args, **kwargs)
         self.node_prealloc_count = node_prealloc_count
 
-        self.preallocated_nodes = test_context.cluster.alloc(
-            ClusterSpec.simple_linux(node_prealloc_count))
+        # Nodes are allocated later on first access
+        self._preallocated_nodes = []
+
+    @property
+    def preallocated_nodes(self):
```

Comment:
nit: it is not immediately from this property, which looks read-only from the name, that it has a side effect of possibly allocating nodes. Although I see it is required to migrate the api from the previous `self.preallocated_nodes` so may be unavoidable

Reply:
Yeah, this could use some refactoring later: pre-allocating nodes was never really the key thing about the tests that subclass this: they really just want to allocate nodes and share them between multiple services.

I think for the moment keeping the naming the same is less disruptive, but a later change could do something like introducing a NodeClaim class that could be created+destroyed to grab these shared nodes, rather than doing it at the level of the test class.